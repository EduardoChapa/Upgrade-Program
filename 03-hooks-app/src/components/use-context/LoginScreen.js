import React, { useContext } from 'react';
import UserContext from './UserContext';

function LoginScreen() {

    const { setUser } = useContext(UserContext);

    return (
        <div>
            <h1>Login Screen</h1>
            <hr/>
            <button onClick={ () => setUser({ id: 999, name: 'Pete', email: 'sample@mail.com' }) }>
                Add user
            </button>
        </div>
    );
}

export default LoginScreen;