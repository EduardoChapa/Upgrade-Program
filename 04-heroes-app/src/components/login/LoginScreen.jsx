import React, { useContext } from 'react';
import { useNavigate } from 'react-router-dom';
import AuthContext from '../../auth/authContext';
import types from '../../types/types';

function LoginScreen() {

    const { dispatch } = useContext(AuthContext);
    const navigate = useNavigate();

    const lastPath = localStorage.getItem('lastPath') || '/marvel';

    const handleSubmit = () => {
        dispatch({
            type: types.login,
            payload: { name: 'Luluis' }
        });
        navigate(lastPath, {
            replace: true
        });
    }

    return (
        <div className='container mt-5'>
            <h1>Login!</h1>
            <hr/>
            <button className='btn btn-primary' onClick={ handleSubmit }>Login</button>
        </div>
    ) 
}

export default LoginScreen;