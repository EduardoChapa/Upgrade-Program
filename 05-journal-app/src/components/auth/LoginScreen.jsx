import React from 'react'
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { googleLogin, emailPasswordLogin } from '../../actions/auth';
import { setError } from '../../actions/ui';
import { useForm } from '../../hooks/useForm';

function LoginScreen() {

  const dispatch = useDispatch();
  const { loading } = useSelector(state => state.ui);

  const [formValues, handleInputChange] = useForm({
    email: 'sample@mail.com',
    password: '1111'
  });

  const { email, password } = formValues;

  const formIsValid = () => {
    if (email.length === 0 || password.length === 0) {
      dispatch(setError('Please fill all the data'));
      return false;
    }
    return true;
  }

  const handleLogin = (e) => {
    e.preventDefault();
    if (formIsValid())
      dispatch(emailPasswordLogin(email, password));
  }

  const handleGoogleLogin = () => {
    dispatch(googleLogin());
  }

  return (
    <div className='animate__animated animate__fadeIn animate__faster'>
      <h3 className='auth__title'>Login</h3>
      <form onSubmit={ handleLogin }>
        <input className='auth__input' type="text" placeholder='Your email...' name='email' autoComplete='off' value={ email } onChange={ handleInputChange }/>
        <input className='auth__input' type="password" placeholder='Your password...' name='password' value={ password } onChange={ handleInputChange }/>
        <button className='btn btn-primary btn-block' type='submit' disabled={ loading }>Login</button>
        <hr/>
        <div className='auth__social-networks'>
          <p>Login with...</p>
          <div className="google-btn" onClick={handleGoogleLogin}>
            <div className="google-icon-wrapper">
              <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
            </div>
            <p className="btn-text">
              <b>Login with Google</b>
            </p>
          </div>
        </div>
        <Link className='link' to='/auth/register'>
          Create new account
        </Link>
      </form>
    </div>
  )
}

export default LoginScreen;