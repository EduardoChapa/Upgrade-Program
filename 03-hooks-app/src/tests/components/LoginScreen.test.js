import React from 'react';
import { mount } from 'enzyme';
import LoginScreen from '../../components/use-context/LoginScreen';
import UserContext from '../../components/use-context/UserContext';

describe('Testing LoginScreen component', () => {

    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{ setUser }} >
            <LoginScreen />
        </UserContext.Provider>
    );

    test('should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should call setUser with the expected argument', () => {
        wrapper.find('button').simulate('click');

        expect(setUser).toHaveReturnedWith({ id: 999, name: 'Pete', email: 'sample@mail.com' });
    });
});