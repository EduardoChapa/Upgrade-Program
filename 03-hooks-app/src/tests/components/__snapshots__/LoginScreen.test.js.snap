// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginScreen component should display correctly 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    onClick={[Function]}
  >
    Add user
  </button>
</div>
`;
